src/app/feature/shared-data/shared-data.service.ts

```diff
@@ -138,6 +149,19 @@ export class SharedDataService extends AsyncServiceBase {
     return this.provider.updateSharedData(id, key, value);
   }
 
+  /**
+   * Update top-level collection properties.
+   * This is used to extend typical shared data properties and should not be used to overwrite
+   * core data (admins, members, metadata etc.)
+   */
+  public setCustomCollectionMeta(id: string, key: string, value: any) {
```

Comment:
nit(non-blocking) I'm a bit confused about the terminology, but isn't the entity at this level a _document_ rather than a _collection_? It sits inside the `shared_data` collection and has a unique ID and has data directly on it, so my understanding is that we should be talking about a `DUMMY_SHARED_DATA_DOCUMENT` and `setCustomCollectionMeta()` method.

Reply:
You're right, it is a document not collection. It get's a bit extra confusing because once in the document we have `doc.data` which firestore refers to as all of the document data, and the `data` that we refer to as being shared data.

So to try disambiguate a little I've updated to `setCustomSharedMeta` to describe everything in a top-level document that is not the child `data` (e.g. members, admins or custom properties such as access_codes)

Updated with [5d6da05](https://github.com/IDEMSInternational/open-app-builder/pull/3044/commits/5d6da050d777720d93aa621893c635ace80f91f9)